global_body_planner/include/global_body_planner/rrt_connect.h

```diff
@@ -12,103 +11,110 @@ using namespace planning_utils;
-     * @param[in] state_sequence The sequence of states in the path
-     * @param[in] action_sequence The sequence of actions in the path
-     * @param[in] terrain Height map of the terrain
-     */
-    void postProcessPath(std::vector<State> &state_sequence, std::vector<Action> &action_sequence, const PlannerConfig &planner_config);
-
-    /**
-     * @brief Post process the path by removing extraneous states that can be bypassed
-     * @param[in] Ta The planning tree originating from the start state
-     * @param[in] Tb The planning tree originating from the end state
-     * @param[out] state_sequence The sequence of states in the path
-     * @param[out] action_sequence The sequence of actions in th  e path     
-     * @param[in] terrain Height map of the terrain
-     */
-    void extractPath(PlannerClass Ta, PlannerClass Tb, std::vector<State> &state_sequence, std::vector<Action> &action_sequence, const PlannerConfig &planner_config);
-
-    /**
-     * @brief Run the full RRT-Connect planner until the goal is found or time has expired, then post process and update statistics
-     * @param[in] terrain Height map of the terrain
-     * @param[in] s_start The start state of the planner
-     * @param[in] s_goal The goal state of the planner
-     * @param[out] state_sequence The sequence of states in the final path
-     * @param[out] action_sequence The sequence of actions in the final path
-     * @param[in] max_planning_time Maximum time allowed to find a plan
-     */
-    void runRRTConnect(const PlannerConfig &planner_config, State s_start, State s_goal, std::vector<State> &state_sequence, std::vector<Action> &action_sequence, double max_planning_time);
-
-  protected:
-
-    /// Time horizon (in seconds) the planner is allowed to search until restarted
-    double anytime_horizon;
-
-    /// Initial guess at how quickly the planner executes (in m/s)
-    const double planning_rate_estimate = 16.0; // m/s (meters planned/computation time)
-    
-    /// Initial anytime horizon (in seconds)
-    double anytime_horizon_init;
-
-    /// Factor by which horizon is increased if replanning is required
-    double horizon_expansion_factor = 1.2;
+class RRTConnectClass : public RRTClass {
+public:
+  /**
+   * @brief Constructor for RRTConnectClass
+   * @return Constructed object of type RRTConnectClass
+   */
+  RRTConnectClass();
+
+  /**
+   * @brief Destructor for RRTConnectClass
+   */
+  ~RRTConnectClass();
+
+  /** Connect the tree to the desired state
+   * @param[in] T The PlannerClass instance containing the tree
+   * @param[in] s The state to connect the tree towards
+   * @param[in] planner_config Configuration parameters
+   * @param[in] direction The direction with which to peform the extension
+   * (FORWARD to go away from the root vertex, REVERSE to go towards it)
+   * @param[in] tree_pub Publisher for broadcasting the tree visual
+   */
+  int connect(PlannerClass &T, State s, const PlannerConfig &planner_config,
```

Comment:
You could also have a pointer to plannerclass in this class that is initialized in the constructor instead of passing as a reference